piwheels/master/the_scribe.py

```diff
@@ -349,6 +361,76 @@ def write_project_page(self, package):
         except FileExistsError:
             pass
 
+    def delete_package(self, package):
+        """
+        Attempts to remove the index and project page directories (including all
+        known wheel files) of the specified *package*.
+
+        :param str package:
+            The name of the package to delete.
+        """
+        if len(package) == 0:
+            # refuse to delete /simple/ and /project/ by accident
+            self.logger.error('not deleting - package name is empty')
+            return
+
+        files = {f for f in self.db.get_package_files(package)}
+
+        for name in ('simple', 'project'):
+            self.logger.info('deleting /%s/ dir for %s', name, package)
+            pkg_dir = self.output_path / 'simple' / package
+            canon_dir = pkg_dir.with_name(canonicalize_name(pkg_dir.name))
+
+            for dir in (pkg_dir, canon_dir):
+                self.logger.debug('Processing %s', dir)
+                index = dir / 'index.html'
+                for file in files:
+                    file_path = dir / file
+                    try:
+                        file_path.unlink()
+                        self.logger.debug('File deleted: %s/%s', dir, file)
+                    except FileNotFoundError:
+                        self.logger.debug('File not found: %s/%s', dir, file)
+
+                try:
+                    index.unlink()
+                    self.logger.debug('Index deleted: %s', index)
+                except FileNotFoundError:
+                    self.logger.debug('File not found: %s', index)
+
+                try:
+                    dir.unlink()
+                except IsADirectoryError:
+                    self.logger.debug('Cannot unlink directory: %s', dir)
+                    try:
+                        dir.rmdir()
+                    except (OSError, NotADirectoryError) as e:
+                        self.logger.debug('Failed to remove directory: %s',
```

Comment:
And again, this a warning. Also, pretty sure `NotADirectoryError` is a subclass of `OSError` :)